etc/scripts/release.hs

```diff
@@ -268,6 +268,24 @@ rules global@Global{..} args = do
                 cmd "strip -o"
                     [out, releaseBinDir </> binaryName </> stackExeFileName]
 
+    releaseDir </> binaryInstallerFileName %> \out -> do
+        need [releaseDir </> binaryExeFileName]
+        need [releaseDir </> binaryInstallerNSIFileName]
+
+        actionOnException
+            (command_ [Cwd releaseDir] "c:\\Program Files (x86)\\NSIS\\Unicode\\makensis.exe"
+                [ "-V3"
+                , binaryInstallerNSIFileName])
+            (removeFile out)
+
+    releaseDir </> binaryInstallerNSIFileName %> \out -> do
+        need ["etc" </> "scripts" </> "build-stack-installer" <.> "hs"]
+        cmd "stack etc/scripts/build-stack-installer.hs" 
+            [ binaryExeFileName
```

Comment:
This ends up creating an installer that installs the executable named e.g. `stack-2.2.0-windows-x86_64-bin.exe`, which means you can't just run `stack` after installing it.  Maybe there's a way to tell the NSI file to rename it to `stack.exe`?  Or maybe you can point the NSI file to `bin\stack-2.2.0-windows-x86_64\stack.exe` instead, which already has the right filename.

Reply:
NSI did provide a oname args to overwrite the output name, I've added it in `build-stack-installer.hs`